students/18th/team1/hurjungwon/account/views.py

```diff
@@ -0,0 +1,73 @@
+import json
+import re
+
+from django.views     import View
+from django.http      import JsonResponse
+from django.db.models import Q
+
+from .models import User
+
+
+class SignUpView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+
+        try:
+            user_name    = data['user_name']
+            email        = data['email']
+            password     = data['password']
+            name         = data.get('name', None)
+            phone_number = data.get('phone_number', None)
+            
+            email_match = re.match('[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[a-zA-Z0-9.]+', email)
+            
+            if not (email_match):
+                return JsonResponse({'message': 'invalid email'}, status=400)
+
+            password_match = re.match('\S{8,}', password)
+            
+            if not(password_match):
+                return JsonResponse({'message': 'invalid password'}, status=400)
+
+            if User.objects.filter(user_name = user_name).exists():
+                return JsonResponse({'message': 'user_name already exists'}, status=400)
```

Comment:
실행하는데 문제도 없고 차이도 없겠지만, 파이썬으로 개발할 때 가독성이 굉장히 중요하기 때문에 물 흐르듯이 읽히는게 중요합니다. 따라서 비슷한 로직 혹은 관련 로직들을 모아주시는게 좋습니다. 아래처럼요!
```suggestion
            email_match = re.match('[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[a-zA-Z0-9.]+', email)
            password_match = re.match('\S{8,}', password)
            
            if not (email_match):
                return JsonResponse({'message': 'invalid email'}, status=400)
            
            if not(password_match):
                return JsonResponse({'message': 'invalid password'}, status=400)

            if User.objects.filter(user_name = user_name).exists():
                return JsonResponse({'message': 'user_name already exists'}, status=400)
```

Reply:
비교해보니까 확실히 가독성이 올라가네요 수정하겠습니다!!!